### ISSUE TITLE
Support for custom role for cross account access

### ISSUE THREAD

**Issue by OlafConijn** [OlafConijn/AwsOrganizationFormation#72]
It *seems* nice to be able to specify a custom IAM role for cross account access.
The role would be specified in the `organization.yml` file on either the level of a OrganizationRoot, OrganizationalUnit or Account Resource.

* The attribute would be called `CrossAccountRoleName`.
* The role is expected to exist in accounts that are contained within the OrganizationRoot, OrganizationalUnit or Account.
* The role is used to set up: `Alias`, `PasswordPolicy` and `SupportLevel` for the account.
* The role is used as a default for cross account access when performing tasks like `update-stacks` or `update-cdk`. The role used, however, can still be overwritten within the Task by specifying a `TaskRoleName`.
* The Role specified is not a custom name for the `OrganizationAccountAccessRole` created when creating the account. The `CrossAccountRoleName` is assumed to be used as a second IAM Role deployed in all accounts in order to minimize/control permissions of the `org-formation` tool. 

example:
``` yaml
  OrganizationRoot:
    Type: OC::ORG::OrganizationRoot
    Properties:
      CrossAccountRoleName: OrgFormationBuildRole # applies to all accounts (within Root)

  SharedOU:
    Type: OC::ORG::OrganizationalUnit
    Properties:
      OrganizationalUnitName: shared
      CrossAccountRoleName: OrgFormationBuildRole # applies to all accounts within OU
      Accounts:
        - !Ref SharedUsersAccount
        - !Ref SharedServicesAccount
        - !Ref SharedComplianceAccount

  SharedUsersAccount:
    Type: OC::ORG::Account
    Properties:
      CrossAccountRoleName: OrgFormationBuildRole  # applies to only this account
      RootEmail: users-2@olafconijn.awsapps.com
      Alias: org-formation-users
      AccountName: Shared Users Account
      PasswordPolicy: !Ref PasswordPolicy
      Tags:
        budget-alarm-threshold: '100'
        account-owner-email: olaf@email.com
```

Question:
* When creating the account, would the a role with the name `CrossAccountRoleName` also be provisioned? would this role be created used `OrganizationAccountAccessRole`? 
  * If not: the build would fail after creating the account and trying to update properties like `Alias`, `PasswordPolicy` and `SupportLevel`.
  * If yes: would the role be provisioned using cloudformation? must the cloudformation be configurable?
 
Workaround:
1. Not specify properties like `Alias`, `PasswordPolicy` and `SupportLevel`.
2. Specify a task that deploys a role used as `TaskRoleName` (see example 1 below)
3. Use the role deployed in step 2 as `TaskRoleName` of other tasks (see example 1 below)
4. Deny assuming the `OrganizationAccountAccessRole` by the identity that executes the org-formation build.  (see example 2 below)
5. If IAM Alias, Password Policy, etc are required use a Custom Resource to implement these.

Workaround example 1: Deploying a IAM Role and using this as TaskRoleName.
* task file: https://github.com/OlafConijn/AwsOrganizationFormation/blob/master/test/integration-tests/resources/scenario-update-stacks-custom-role/1-deploy-stacks-custom-roles.yml
* roles: https://github.com/OlafConijn/AwsOrganizationFormation/blob/master/test/integration-tests/resources/scenario-update-stacks-custom-role/my-roles.yml

Workaround example 2: deny assuming the OrganizationAccountAccessRole
``` yaml
Effect: Deny
Action: 'sts:AssumeRole'
Resource: 'arn:aws:iam::*:role/OrganizationAccountAccessRole'
```

Question:
* Would the workaround be preferable over the `CrossAccountRoleName`? As without the workaround it would not be possible explicitly deny the of `OrganizationAccountAccessRole`.
* If workaround is preferred, does it make sense to have a default TaskRoleName that can be passed to the `perform-tasks` command or configured on `include` tasks?

other thoughts?

**Comment by OlafConijn** [comment]
Added the `OrganizationAccessRoleName` property to `OC::ORG::Account`.

The `OrganizationAccessRoleName` will be used to access information and make changes to the account. When not specified the default role name from the `OC::ORG::OrganizationRoot` will be used. If none specified the default role `OrganizationAccountAccessRole` is used.

When creating an account with an `OrganizationAccessRoleName` specified the role will be created as part of the account creation process.

When chancing the `OrganizationAccessRoleName` no roles will be renamed. The tool will expect you to ensure the roles are present.

**Comment by OlafConijn** [comment]
will be part of v.0.9.6.
currently abvailable in 0.9.6-beta: https://www.npmjs.com/package/aws-organization-formation/v/0.9.6-beta.6
commit: 74c9b7374769500c1e7107fc9570d57fe919056d

**Comment by OlafConijn** [comment]
released as part of v0.9.6
